amethyst_rendy/src/camera.rs

```diff
@@ -49,7 +61,12 @@ impl Orthographic {
         matrix[(1, 3)] = -(top + bottom) / (top - bottom);
         matrix[(2, 3)] = -z_near / (z_far - z_near);
 
-        Self { matrix }
+        Self {
+            matrix,
+            inverse_matrix: matrix
```

Comment:
While singular projection matrix probably never occurs for correct use cases, I'd prefer deferring this error (panic) until the user perform any operation that actually requires the inverse matrix.

Reply:
I'd argue this is an unnecessary complication. All matrix based cameras must have an invertible matrix. Having to check that lazily has also additional cost. There are multiple places where camera inverse is necessary, so having it precomputed is very sensible in my opinion.